Moved 'Missing Props Validation' error info to earlier lesson

remel-crowe opened the pull request:
<!-- Thank you for taking the time to contribute to The Odin Project. In order to get this pull request (PR) merged in a reasonable amount of time, you must complete this entire template. -->

## Because
<!-- Summarize the purpose or reasons for this PR, e.g. what problem it solves or what benefit it provides. -->
Moving the information about  'Missing Props Validation' error to an earlier lesson to aid with the flow of the section. Users coding along will likely face this issue before the this lesson and will need to know how to navigate it.


## This PR
<!-- A bullet point list of one or more items describing the specific changes. -->
- Deleted informational div from [react/passing_data_between_components.md](https://github.com/TheOdinProject/curriculum/compare/main...remel-crowe:curriculum:move_props_error?expand=1#diff-1e80ba6c4b4361c8b9279d15ccb58c1cf13a96f65e80f3f6ce6fa362dbee4252)
- Added informational div to [react/rendering_techniques.md](https://github.com/TheOdinProject/curriculum/compare/main...remel-crowe:curriculum:move_props_error?expand=1#diff-d1d84d9834126168b0514efb09eceaa2ffcf312870902a1e6d532853ebf15a62)

## Issue
<!--
If this PR closes an open issue in this repo, replace the XXXXX below with the issue number, e.g. Closes #2013.

If this PR closes an open issue in another TOP repo, replace the #XXXXX with the URL of the issue, e.g. Closes https://github.com/TheOdinProject/curriculum/issues/XXXXX

If this PR does not close, but is related to another issue or PR, you can link it as above without the 'Closes' keyword, e.g. 'Related to #2013'.

_Note:_ any pull request created for an issue that already has someone else assigned **will be closed without review**.
-->
Closes #27409

## Additional Information
<!-- Any other information about this PR, such as a link to a Discord discussion. -->


## Pull Request Requirements
<!-- Replace the whitespace between the square brackets with an 'x', e.g. [x]. After you create the PR, they will become checkboxes that you can click on. -->
-   [ x] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/.github/blob/main/CONTRIBUTING.md)
-   [ x] The title of this PR follows the `location of change: brief description of change` format, e.g. `Intro to HTML and CSS lesson: Fix link text`
-   [ x] The `Because` section summarizes the reason for this PR
-   [x ] The `This PR` section has a bullet point list describing the changes in this PR
-   [ x] If this PR addresses an open issue, it is linked in the `Issue` section
-   [ x] If any lesson files are included in this PR, they have been previewed with the [Markdown preview tool](https://www.theodinproject.com/lessons/preview) to ensure it is formatted correctly
-   [ x] If any lesson files are included in this PR, they follow the [Layout Style Guide](https://github.com/TheOdinProject/curriculum/blob/main/LAYOUT_STYLE_GUIDE.md)

wise-king-sullyman reviewed (approved):
Looks great, thanks!

remel-crowe commented:
Hi Sullyman! Thanks for the merge. I've just noticed I moved it but didn't actually change the name of the prop the tip refers to. Should I create another PR for this? I've fixed the error.
